src/Agent.Worker/Variables.cs

```diff
@@ -589,6 +613,58 @@ public void CopyInto(Dictionary<string, VariableValue> target)
             }
         }
 
+        private Boolean IsReadOnly(Variable variable)
+        {
+            if (variable.ReadOnly)
+            {
+                return true;
+            }
+
+            if (_wellKnownSystemVariables == null)
+            {
+                _wellKnownSystemVariables = new List<String>();
+
+                // If variable is not marked as readOnly, return whether its in the set of reserved system variables (which are always readOnly)
+                List<String> readOnlyPrefixes = new List<String>()
```

Comment:
nit: I would make this 

`private IReadOnlyList<String> readOnlyPrefixes = new List<String>() `
declared outside the function.